research/build-reverse-dependencies/index.md

```diff
@@ -0,0 +1,98 @@
+- [copr-autorebuilder](https://pagure.io/copr-autorebuilder/tree/master)
+- [copr-rebuild-tools](https://github.com/fedora-copr/copr-rebuild-tools/tree/main)
+
+The first two in the above list are specifically designed for resolving reverse dependencies and this research will focus on them.
+
+## Koschei
+
+[Koschei](https://apps.fedoraproject.org/koschei) is a **continuous integration service** for RPM packages that tracks package dependency changes in Fedora Rawhide and rebuilds packages whose dependencies change too much.
+
+**Koschei depends on Koji**. New Koji builds "trigger" Koschei and Koschei rebuilds reverse dependencies as Koji scratch builds.
+**Koji scratch builds do not trigger Koshei** at the moment.
+This means that Koschei can be used as a CI service only after a _downstream release synchronization_ but **our users would like to check changes before making a downstream release synchronization.**
+
+If we want to use Koshei as an upstream CI service (or even downstream but without the need of a downstream release synchronization) we need to **make it react to scratch builds**.
+
+The Koschei project is not much lively, at least upstream; different reported issues are not being answered or worked on. For this reason I fear integrating new features in the Koschei production service could take a long time.
+
+If we want to use Koschei we could also think about **fork the project** and create our own instance inside the Packit cluster. This approach could give us two advantages:
+
+- Shorter time to production and our changes can always be taken back by the original project at some point.
+- We can re-enable COPR as a backend for Koschei builds. Accordingly to [this thread](https://pagure.io/fedora-ci/general/issue/45) Koschei has had a COPR backend for a while, but it has been removed (because COPR is a third party service not supported by the Fedora Infrastructure). Enabling COPR as a backend will allow the service to work even for those upstream projects which are not completely integrated in Fedora (if needed).
+
+What are the **PROs** I see in re-using Koshei:
+
+- We could benefit from an already well known somehow "standardized" user interface (the Koschei frontend).
+- We shouldn't need to be able to calculate reverse dependencies in Packit and submit builds from Packit if we will reuse (the Koschei backend).
+
+What are the **CONs** I see in re-using Koshei:
+
+- As far as I can understand it, the guy who wrote Koshei does not work anymore in Red Hat. We can not ask him for any help dealing with the Koschei code.
+- We could face unknown problems.
+
+## Mass Prebuild
+
+This is a CLI tool that resolves reverse dependencies and rebuild them all. It can work with both Koji and COPR.
+Even if it is a CLI tool we can not easily integrate it in packit-service because of two main design choices:
+
+- the tool interacts with a database
+- the tool keeps a session and can interact with the user
```

Comment:
so there is no way of disabling this interactive mode?

Reply:
yes, you can decide to submit one step at time, specifying which step you want, but still the code is highly coupled with the "entire session idea" and the database.

Reply:
understood :+1: